intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/problems/MinorWarnings.kt

```diff
@@ -201,4 +201,26 @@ class UnknownServiceClientValue(descriptorPath: String, serviceClient: String) :
 ) {
   override val level
     get() = Level.WARNING
+}
+
+/**
+ * @see [K2IdeModeCompatibilityVerifier]
+ */
+private const val UNDECLARED_KOTLIN_K2_COMPATIBILITY_MODE_MESSAGE = "Plugin depends on the Kotlin plugin (org.jetbrains.kotlin) but does not declare " +
+  "a compatibility mode in the <supportsKotlinPluginMode> extension. " +
+  "This feature is available in the IntelliJ IDEA 2024.2.1 or later."
+
+data class UndeclaredKotlinK2CompatibilityMode(val descriptorPath: String) : InvalidDescriptorProblem(
+  descriptorPath = descriptorPath,
+  detailedMessage = UNDECLARED_KOTLIN_K2_COMPATIBILITY_MODE_MESSAGE
+) {
+  val detailedMessage = UNDECLARED_KOTLIN_K2_COMPATIBILITY_MODE_MESSAGE
+
+  override val level
+    get() = Level.WARNING
+
+  override val hint = ProblemSolutionHint(
```

Comment:
AFAIU this is not shown anywhere yet? Shouldn't we add the link to the problem message then?

Reply:
Yes, it is not shown anywyhere now. This problem from the `structure` library is remapped to the `UndeclaredKotlinK2CompatibilityModeProblem` in the Plugin Verifier. 

I have added the link to the problem message to the Plugin Verifier side to make it visible in the reports.